src/Uno.UI/UI/Xaml/Controls/MediaPlayerElement/MediaTransportControls.MediaPlayer.cs

```diff
@@ -295,84 +372,72 @@ private void ToggleMute(object sender, RoutedEventArgs e)
 			ResetControlsVisibilityTimer();
 		}
 
-#if __ANDROID__ || __IOS__ || __MACOS__
 		private void ThumbOnDragStarted(object sender, DragStartedEventArgs dragStartedEventArgs)
 		{
-			if (_mediaPlayer != null && !_isScrubbing)
+			if (_mediaPlayer != null && !m_isInScrubMode)
 			{
 				_wasPlaying = _mediaPlayer.PlaybackSession.IsPlaying;
-				_skipPlayPauseStateUpdate = true;
-				_isScrubbing = true;
+				EnterScrubbingMode();
 
 				_mediaPlayer.Pause();
 			}
 		}
-#else
-		private void OnPointerEntered(object sender, PointerRoutedEventArgs e)
-		{
-			if (_mediaPlayer != null && !_isScrubbing)
-			{
-				_wasPlaying = _mediaPlayer.PlaybackSession.IsPlaying;
-				_isScrubbing = true;
-			}
-		}
-
-		private void OnPointerExited(object sender, PointerRoutedEventArgs e)
-		{
-			_isScrubbing = false;
-			if (_wasPlaying)
-			{
-				_mediaPlayer?.Play();
-			}
-		}
-#endif
 
 		private void ThumbOnDragCompleted(object sender, DragCompletedEventArgs dragCompletedEventArgs)
 		{
 			if (m_tpMediaPositionSlider is null || double.IsNaN(m_tpMediaPositionSlider.Value))
 			{
 				return;
 			}
+
 			if (_mediaPlayer != null)
 			{
-				if (_mediaPlayer.PlaybackSession.IsPlaying)
+				if (m_isPlaying || m_isBuffering)
 				{
-					_skipPlayPauseStateUpdate = true;
-					_isScrubbing = true;
-					_wasPlaying = true;
+					EnterScrubbingMode();
 					_mediaPlayer.Pause();
 				}
+
 				_mediaPlayer.PlaybackSession.Position = TimeSpan.FromSeconds(m_tpMediaPositionSlider.Value);
+
 				if (_wasPlaying)
 				{
 					_mediaPlayer.Play();
 				}
 			}
-			_isScrubbing = false;
-			_skipPlayPauseStateUpdate = false;
+
+			ExitScrubbingMode();
 		}
 
-		public void TappedProgressSlider(object sender, RoutedEventArgs e)
+		private void EnterScrubbingMode()
 		{
-			_isScrubbing = true;
-			if (m_tpMediaPositionSlider is null || double.IsNaN(m_tpMediaPositionSlider.Value))
+			if (m_transportControlsEnabled && _mediaPlayer is { })
 			{
-				_isScrubbing = false;
-				return;
+				if (//!m_isAudioOnly &&
+					//!IsLiveContent() &&
+					!m_isInScrubMode)
+				{
+					m_currentPlaybackRate = _mediaPlayer.PlaybackSession.PlaybackRate;
+					_mediaPlayer.PlaybackSession.PlaybackRate = 0;
+					//IFC(EnableValueChangedEventThrottlingOnSliderAutomation(false));
+					m_isInScrubMode = true;
+				}
 			}
-			if (_mediaPlayer != null)
+		}
+
+		private void ExitScrubbingMode()
+		{
+			if (m_transportControlsEnabled && _mediaPlayer is { })
 			{
-				_wasPlaying = _mediaPlayer.PlaybackSession.IsPlaying;
-				_skipPlayPauseStateUpdate = true;
-				_mediaPlayer.Pause();
-				_mediaPlayer.PlaybackSession.Position = TimeSpan.FromSeconds(m_tpMediaPositionSlider.Value);
-				if (_wasPlaying)
+				if (//!m_isAudioOnly &&
+					//!IsLiveContent() &&
```

Comment:
It's probably best to exclude those blocks this way, to signify that it comes from not implemented features.
```suggestion
				if (
#if !HAS_UNO
					!m_isAudioOnly &&
					!IsLiveContent() &&
#endif
```